lib/timers.js

```diff
@@ -434,6 +482,15 @@ function rearm(timer) {
 
 
 const clearTimeout = exports.clearTimeout = function(timer) {
+  // Fewer checks may be possible, but these cover everything.
+  if (async_hook_fields[kDestroy] > 0 &&
+      timer &&
+      typeof timer[async_id_symbol] === 'number' &&
+      !timer._destroyed) {
+    emitDestroy(timer[async_id_symbol]);
+    timer._destroyed = true;
+  }
```

Comment:
Ok well, the tests seem to ignore timeouts in the net tests so it was never caught. Run any net server connection and you will quickly find that destroy is never reported on timers for any connection that does not timeout.